helpers/helpers.v2.1.d/utils

```diff
@@ -456,3 +456,148 @@ ynh_spawn_app_shell() {
+	elif [ $usable_space -ge $(( $swap_max_size / 4 )) ]
+	then
+		local swap_size=$(( $swap_max_size / 4 ))
+	else
+		echo "Not enough space left for a swap file" >&2
+		local swap_size=0
+	fi
+
+	# If there's enough space for a swap, and no existing swap here
+	if [ $swap_size -ne 0 ] && [ ! -e /swap_$app ]
+	then
+		# Create file
+		truncate -s 0 /swap_$app
+		
+		# set the No_COW attribute on the swapfile with chattr
+		chattr +C /swap_$app
+		
+		# Preallocate space for the swap file, fallocate may sometime not be used, use dd instead in this case
+		if ! fallocate -l ${swap_size}K /swap_$app
+		then
+			dd if=/dev/zero of=/swap_$app bs=1024 count=${swap_size}
+		fi
+		chmod 0600 /swap_$app
+		# Create the swap
+		mkswap /swap_$app
+		# And activate it
+		swapon /swap_$app
+		# Then add an entry in fstab to load this swap at each boot.
+		echo -e "/swap_$app swap swap defaults 0 0 #Swap added by $app" >> /etc/fstab
+	fi
+}
+
+ynh_del_swap () {
+	# If there a swap at this place
+	if [ -e /swap_$app ]
+	then
+		# Clean the fstab
+		sed -i "/#Swap added by $app/d" /etc/fstab
+		# Desactive the swap file
+		swapoff /swap_$app
+		# And remove it
+		rm /swap_$app
+	fi
+}
+
+# Check if the device of the main mountpoint "/" is an SD card
+#
+# [internal]
+#
+# return 0 if it's an SD card, else 1
+ynh_is_main_device_a_sd_card () {
+  if [ "$(systemd-detect-virt)" != "none" ]; then
+    # Assume virtualization does not take place on SD card
+    return 1
+  fi
+
+	local main_device=$(lsblk --output PKNAME --noheadings $(findmnt / --nofsroot --uniq --output source --noheadings --first-only))
+
+	if echo $main_device | grep --quiet "mmc" && [ $(tail -n1 /sys/block/$main_device/queue/rotational) == "0" ]
+	then
+		return 0
+	else
+		return 1
+	fi
+}
+
+# Check available space before creating a temp directory.
+#
+# usage: ynh_smart_mktemp --min_size="Min size"
+#
+# | arg: -s, --min_size= - Minimal size needed for the temporary directory, in Mb
+ynh_smart_mktemp () {
+    # Declare an array to define the options of this helper.
+    declare -Ar args_array=( [s]=min_size= )
+    local min_size
+    # Manage arguments with getopts
+    ynh_handle_getopts_args "$@"
+
+    min_size="${min_size:-300}"
+    # Transform the minimum size from megabytes to kilobytes
+    min_size=$(( $min_size * 1024 ))
+
+    # Check if there's enough free space in a directory
+    is_there_enough_space () {
+            local free_space=$(df --output=avail "$1" | sed 1d)
+            test $free_space -ge $min_size
+    }
+
+    if is_there_enough_space /tmp; then
+            local tmpdir=/tmp
+    elif is_there_enough_space /var; then
+            local tmpdir=/var
+    elif is_there_enough_space /; then
+            local tmpdir=/   
+    elif is_there_enough_space /home; then
+            local tmpdir=/home
+    else
+            ynh_die "Insufficient free space to continue..."
+    fi
+
+    echo "$(sudo mktemp --directory --tmpdir="$tmpdir")"
```

Comment:
Not sure why there was a sudo here in one of the variants...

Reply:
```suggestion
    echo "$(mktemp --directory --tmpdir="$tmpdir")"
```